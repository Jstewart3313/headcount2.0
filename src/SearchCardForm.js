import React, { Component } from 'react'

class SearchCardForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: ''
    }
  }

  trackChange = (event) => {
    const { value } = event.target;
    this.setState({location: value})
    this.props.searchCard(this.state.location)
  }

  render() {
    return(
      <div>
      <form 
        classname='search-form'
        onSubmit=''
      >
        <input 
          value={this.state.location}
          placeholder='Search'
          name= 'location'
          onChange={this.trackChange}
        />
      </form>
      </div>
    )
  }











}

export default SearchCardForm
