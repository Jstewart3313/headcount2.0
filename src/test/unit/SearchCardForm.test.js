import React from 'react';
import { shallow, mount } from 'enzyme';

import SearchCardForm from '../../SearchCardForm';

describe( 'SearchCardForm', () => {
  

  it('should match the snapshop', () => {

    const mockSearchCard = jest.fn()
    const wrapper = shallow(<SearchCardForm searchCard={mockSearchCard} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call trackChange when search input is changed', () => {
    const mockSearchCard = jest.fn()
    const wrapper = shallow(<SearchCardForm searchCard={mockSearchCard} />)
    const spy = spyOn(wrapper.instance(), 'trackChange')
    wrapper.instance().forceUpdate()
    const mockEvent = { target: { value: 'Colorado' } }

    wrapper.find('.search-input').simulate('change', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('updates state when trackChange is called', () => {
  const mockSearchCard = jest.fn()
  const wrapper = shallow(<SearchCardForm searchCard={mockSearchCard} />)
  const mockEvent = { target: { name: 'location', value: 'Colorado' } }

  wrapper.instance().trackChange(mockEvent)

  expect(wrapper.state('location')).toBe('Colorado')
  })

  it('calls handleSubmit onSubmit of the form', () => {
    const mockSearchCard = jest.fn()
    const wrapper = shallow(<SearchCardForm searchCard={mockSearchCard} />)
    const spy = spyOn(wrapper.instance(), 'handleSubmit');
    const mockEvent = { preventDefault: jest.fn() }
    wrapper.instance().forceUpdate();

    wrapper.find('.search-form').simulate('submit', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('calls searchCard when trackChange is called', () => {
    const mockSearchCard = jest.fn()
    const wrapper = shallow(<SearchCardForm searchCard={mockSearchCard} />)
    const mockEvent = { target: { value: 'Colorado' } }
    wrapper.instance().trackChange(mockEvent)

    expect(mockSearchCard).toHaveBeenCalled()
  })







})





