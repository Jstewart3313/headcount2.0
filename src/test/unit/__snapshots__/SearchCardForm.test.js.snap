// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCardForm should match the snapshop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchCardForm
    searchCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="search-form"
        onSubmit={[Function]}
>
        <input
                className="search-input"
                name="location"
                onChange={[Function]}
                placeholder="Search"
                value=""
        />
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-input"
          name="location"
          onChange={[Function]}
          placeholder="Search"
          value=""
/>,
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "name": "location",
          "onChange": [Function],
          "placeholder": "Search",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="search-form"
          onSubmit={[Function]}
>
          <input
                    className="search-input"
                    name="location"
                    onChange={[Function]}
                    placeholder="Search"
                    value=""
          />
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="search-input"
            name="location"
            onChange={[Function]}
            placeholder="Search"
            value=""
/>,
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "name": "location",
            "onChange": [Function],
            "placeholder": "Search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
